rpcd/playbooks/gen-facts.yml

```diff
@@ -0,0 +1,18 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Generate facts
+  hosts: "{{ hosts | default('mons') }}"
```

Comment:
It's working because 'hosts' in that context applies as a variable name. If it were to be quoted, ie 'hosts' then it would be a string and the play would always run against all hosts.